### PR TITLE
Rework automatic code samples in docstrings

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR reworks the automatic code sample docstrings in two ways:

First, use the auto-classes for the preprocessing. As was decided internally, we want to document the model class used, but use the auto classes for preprocessing so users are not confused when a given model uses the tokenizer/feature extractor/image processor/processor of another.

Second, we don't want to showcase `hf-internal-testing` models in the docstrings. Those are tiny random models and it confuses users more than it helps. However when using the standard checkpoint we get doctest problems, so this PR removes the output/loss from the code example when it shouldn't be tested.

Two examples are shown with BERT and DeBERTaV2, I can add more models to the PR if it suits everyone.

### CODE REVIEW

File: src/transformers/utils/doc.py [hunk 1]
```diff
@@ -252,44 +246,37 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
     >>> inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
 
     >>> with torch.no_grad():
     ...     logits = model(**inputs).logits
 
-    >>> predicted_class_id = logits.argmax().item()
-    >>> model.config.id2label[predicted_class_id]
-    {expected_output}
-    ```
+    >>> predicted_class_ids = torch.arange(0, logits.shape[-1])[torch.sigmoid(logits).squeeze() > 0.5]
```
Comment (sgugger) [inline src/transformers/utils/doc.py:259]: Here the code sample did not make any sense for multi-label classification, so changed it.

File: src/transformers/utils/doc.py [hunk 2]
```diff
@@ -543,17 +524,17 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
-    >>> from transformers import {processor_class}, {model_class}
+    >>> from transformers import ImageProcessor, {model_class}
     >>> import torch
     >>> from datasets import load_dataset
 
     >>> dataset = load_dataset("huggingface/cats-image")
     >>> image = dataset["test"]["image"][0]
 
-    >>> feature_extractor = {processor_class}.from_pretrained("{checkpoint}")
+    >>> image_processor = ImageProcessor.from_pretrained("{checkpoint}")
```
Comment (sgugger) [inline src/transformers/utils/doc.py:534]: Adapted names in the vision examples.

File: src/transformers/models/bert/modeling_bert.py [hunk 3]
```diff
@@ -1184,7 +1183,6 @@ def set_output_embeddings(self, new_embeddings):
 
     @add_start_docstrings_to_model_forward(BERT_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
     @add_code_sample_docstrings(
-        processor_class=_TOKENIZER_FOR_DOC,
         checkpoint=_CHECKPOINT_FOR_DOC,
```
Comment (ydshieh) [inline src/transformers/models/bert/modeling_bert.py:1186]: not sure why this is not added say as the last argument and same question below

**Review by ydshieh — APPROVED** [review]
LGTM, but maybe I miss sth - why for most of them, we don't add `processor_class="AutoTokenizer",`

**Review by ArthurZucker — APPROVED** [review]
LGTM it simplifies a lot for addition of specific heads without checkpoints. 
Are you planning on also updating previously added models ?

File: src/transformers/utils/doc.py [hunk 4]
```diff
@@ -543,17 +524,17 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
     Example:
 
     ```python
-    >>> from transformers import {processor_class}, {model_class}
+    >>> from transformers import ImageProcessor, {model_class}
```
[the remark this answers is not in the document]
Comment (sgugger) [reply src/transformers/utils/doc.py:527]: Just a typo, I forgot the Auto here.

File: src/transformers/models/bert/modeling_bert.py (hunk 3 above)
Comment (sgugger) [reply src/transformers/models/bert/modeling_bert.py:1186]: Because there is no `processor_class` in the code sample used by this model anymore. It's just in the base model, and I have just left it there in case it's used by other modalities.

**Review by amyeroberts — APPROVED** [review]
Thanks for adding! Looks a lot tidier 🧹 🧹 🧹

**Review by LysandreJik — APPROVED** [review]
Nice! Thanks!
